test/tests/functional/pbs_job_array.py

```diff
@@ -165,12 +165,26 @@ def test_running_subjob_survive_restart(self):
         self.server.expect(JOB, {'job_state': 'X'},
                            j.create_subjob_id(j_id, 1), max_attempts=1)
 
-        # 6. subjob 2 should be R
-        self.server.expect(JOB, {'job_state': 'R'}, subjid_2, max_attempts=1)
+        subjob2_finished = False
+        # 6. subjob 2 should be in R or X state
+        try:
+            self.server.expect(JOB, {'job_state': 'R'}, subjid_2,
+                               max_attempts=1)
+        except PtlExpectError:
+            self.server.expect(JOB, {'job_state': 'X'}, subjid_2,
```

Comment:
I think by doing this you are changing the intent of the test. If server has a bug where it starts considering running subjobs as exitied upon a restart then this test will never catch that.

IMO it is better to increase the sleep time.

Reply:
I had reservations about this, but I think you are right. After going through the documentation I found this

> Subjob exit status is available only as long as the subjob is in job history. When a subjob is not in job history, a failed or terminated subjob will show an exit status of Finished, instead of failed or terminated.

Would it be better to verify the exit status as zero of that subjob from the server logs? Or checking the exit_status is also not a good idea? What do you think? I just want to avoid increasing the sleep as much as possible, so that in slower machines it does not fail again.

Reply:
I wouldn't rely on exit status either, as that can change in future. How about we increase the sleep time of the job and not wait for the first subjob to finish.  In this test, step 2 where we wait for second subjob to start really serves no purpose. 
If we restart the server as soon as we find the first subjob running, that is all we need. Doing this will speed up the test anyway, since we are not going to wait for a subjob to finish.